gpytorch/kernels/kernel.py

```diff
@@ -320,37 +320,17 @@ def covar_dist(
             x2 = x2.transpose(-1, -2).unsqueeze(-1)
 
         x1_eq_x2 = torch.equal(x1, x2)
-
-        # torch scripts expect tensors
-        postprocess = torch.tensor(postprocess)
-
         res = None
 
-        # Cache the Distance object or else JIT will recompile every time
-        if not self.distance_module or self.distance_module._postprocess != dist_postprocess_func:
-            self.distance_module = Distance(dist_postprocess_func)
-
         if diag:
             # Special case the diagonal because we can return all zeros most of the time.
             if x1_eq_x2:
-                res = torch.zeros(*x1.shape[:-2], x1.shape[-2], dtype=x1.dtype, device=x1.device)
-                if postprocess:
-                    res = dist_postprocess_func(res)
-                return res
+                return torch.zeros(*x1.shape[:-2], x1.shape[-2], dtype=x1.dtype, device=x1.device)
             else:
-                res = torch.norm(x1 - x2, p=2, dim=-1)
-                if square_dist:
-                    res = res.pow(2)
-            if postprocess:
-                res = dist_postprocess_func(res)
-            return res
-
-        elif square_dist:
-            res = self.distance_module._sq_dist(x1, x2, postprocess, x1_eq_x2)
+                res = torch.linalg.norm(x1 - x2, dim=-1)  # 2-norm by default
+                return res.pow(2) if square_dist else res
         else:
-            res = self.distance_module._dist(x1, x2, postprocess, x1_eq_x2)
-
-        return res
+            return (_sq_dist if square_dist else _dist)(x1, x2, x1_eq_x2)
```

Comment:
I find this kind of hard to read, but that might be personal taste
```suggestion
            distfunc = _sq_dist if square_dist else _dist
            return distfunc(x1, x2, x1_eq_x2)
```